src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/experimental/FlatFW.java

```diff
@@ -0,0 +1,249 @@
+        string1().set(value, StandardCharsets.UTF_8);
+        limit(string1().build().limit());
+      }
+      return this;
+    }
+
+    public Builder string1(StringFW value) {
+      prepareToSetField(INDEX_STRING1);
+      StringFW.Builder $string1 = string1();
+      $string1.set(value);
+      limit($string1.build().limit());
+      return this;
+    }
+
+    public Builder string1(DirectBuffer buffer, int offset, int length) {
+      prepareToSetField(INDEX_STRING1);
+      StringFW.Builder string1 = string1();
+      string1.set(buffer, offset, length);
+      limit(string1.build().limit());
+      return this;
+    }
+
+    private int fixed3() {
+      return buffer().getInt(string1().limit() + FIELD_OFFSET_FIXED3);
+    }
+
+    public Builder fixed3(int value) {
+      prepareToSetField(INDEX_FIXED3);
+      buffer().putInt(string1().build().limit() + FIELD_OFFSET_FIXED3, value);
+      return this;
+    }
+
+    private StringFW.Builder string2() {
+      int anchor = string1().build().limit() + FIELD_OFFSET_STRING2;
+      return string2RW.wrap(buffer(), anchor, maxLimit());
+    }
+
+    public Builder string2(String value) {
+      prepareToSetField(INDEX_STRING2);
+      if (value == null) {
+        limit(string1().build().limit() + FIELD_OFFSET_STRING2);
+      } else {
+        string2().set(value, StandardCharsets.UTF_8);
+        limit(string2().build().limit());
+      }
+      return this;
+    }
+
+    public Builder string2(StringFW value) {
+      prepareToSetField(INDEX_STRING2);
+      StringFW.Builder $string2 = string2();
+      $string2.set(value);
+      limit($string2.build().limit());
+      return this;
+    }
+
+    public Builder string2(DirectBuffer buffer, int offset, int length) {
+      prepareToSetField(INDEX_STRING2);
+      StringFW.Builder string2 = string2();
+      string2.set(buffer, offset, length);
+      limit(string2.build().limit());
+      return this;
+    }
+
+    @Override
+    public Builder wrap(MutableDirectBuffer buffer, int offset, int maxLimit) {
+      fieldsSet.clear();
+      super.wrap(buffer, offset, maxLimit);
+      return this;
+    }
+
+    @Override
+    public FlatFW build()
+    {
+        setDefaults();
+        for (int i=0; i < FIELD_COUNT; i++) {
+            if (!fieldsSet.get(i))
+            {
+                throw new IllegalStateException(format("Required field \"%s\" is not set", FIELD_NAMES[i]));
+            }
+        }
+        fieldsSet.clear();
+        return super.build();
+    }
+
+    private void prepareToSetField(int index) {
+        if (fieldsSet.get(index))
+        {
+            throw new IllegalStateException(format("Field \"%s\" has already been set", FIELD_NAMES[index]));
+        }
+        for (int i=0; i < index; i++) {
+            if (!fieldsSet.get(i) && !FIELDS_WITH_DEFAULTS.get(i)) {
+                throw new IllegalStateException(format("Required field \"%s\" is not set", FIELD_NAMES[i]));
+            }
+        }
```

Comment:
Replace `for` loop with `BitSet.nextClearBit(...)` followed by same check as all-fields-set in `build()` but limited by index?
```
private void checkRequiredFieldsSet(
  int fromIndex,
  int toIndex)
{
  int fieldNotSet = fromIndex - 1;
  do
  {
    fieldNotSet = fieldsSet.nextClearBit(fieldNotSet + 1);
  } while (fieldNotSet < toIndex && FIELDS_WITH_DEFAULTS.get(fieldNotSet));

  if (fieldNotSet < toIndex)
  {
    throw new IllegalStateException(format("Required field \"%s\" is not set", FIELD_NAMES[fieldNotSet]));
  }
}
```

Reply:
```
private void checkFieldNotSet(
  int index)
{
  if (fieldsSet.get(index))
  {
    throw new IllegalStateException(format("Field \"%s\" has already been set", FIELD_NAMES[index]));
  }
}
```

Reply:
Done